### PR TITLE
Scrum 48 nest be implement matching algorithm

### PR DESCRIPTION
**Pull request by Mohammed87FS**
# PR: Matching Algorithm Implementation

## Summary
Matching algorithm that calculates compatibility scores based on tags, priorities, and semantic similarity scores.

## Features
- Tag-based matching with priority weighting
- Semantic similarity score fetching between related fields
- Efficient caching system with 24-hour expiration
- Normalized scoring (0-1 range)

## Technical Details
- Endpoint at GET /match/:userId for retrieving matches by user id of a student
- Created `MatchingService` with bidirectional similarity cache (example : ai to machine learning - machine learning to ai)
- Added unit tests
- Handled edge cases (missing tags, priority changes)
- Optimized API calls with prefetching

### CODE REVIEW

File: backend/src/modules/matching/entity/match.entity.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class Match {
+  @ApiProperty({
+    description: 'The unique identifier of the student in this match',
+    example: '5f8d0d55b54764421b7156f1',
```
Comment (jb-cc) [inline backend/src/modules/matching/entity/match.entity.ts:6]: I would maybe change the example to a real UUID (if it is one) to make it clearer :)

can take this one or something idk `12ff5f65-b022-4040-883e-e1045c46ca7c`

File: backend/src/modules/matching/entity/match.entity.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  })
+  studentId: string;
+
+  @ApiProperty({
+    description: 'The unique identifier of the supervisor in this match',
+    example: '5f8d0d55b54764421b7156f2',
```
Comment (jb-cc) [inline backend/src/modules/matching/entity/match.entity.ts:12]: same here

File: backend/src/modules/matching/entity/match.entity.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  })
+  supervisorId: string;
+
+  @ApiProperty({
+    description: 'Compatibility score between student and supervisor (0-1)',
+    example: 85,
```
Comment (jb-cc) [inline backend/src/modules/matching/entity/match.entity.ts:18]: if the minimum is 0, and the maximum is 1, how can it return 85?😅

Comment (Mohammed87FS) [reply backend/src/modules/matching/entity/match.entity.ts:18]: Oh.. Will correct that !

**Review by jb-cc — APPROVED** [review]
The comments I left are not necessary to fix right now. I will approve this as the code looks like it should work (crazy logic btw). Please make sure you tested this on your machine. If you did, and @bernhaaard also approves, you can merge this. 

Good job!

**Review by bernhaaard — APPROVED** [review]
Looks good!
